prefect/deployment.py

```diff
@@ -0,0 +1,144 @@
+import os
+import re
+import sys
+import logging
+import requests
+import pandas as pd
+from io import StringIO
+from dotenv import load_dotenv
+from datetime import datetime, timedelta, timezone
+from dune_client.client import DuneClient
+
+deployments_path = os.path.abspath("/deployments")
+if deployments_path not in sys.path:
+    sys.path.insert(0, deployments_path)
+
+from typing import Any
+from prefect import flow, task, get_run_logger
+from prefect.deployments import Deployment
+from prefect_github.repository import GitHubRepository
+
+from src.models.block_range import BlockRange
+from src.fetch.orderbook import OrderbookFetcher
+from src.models.order_rewards_schema import OrderRewards
+
+load_dotenv()
+
+def get_last_monday_midnight_utc():
+    now = datetime.now(timezone.utc)
+    current_weekday = now.weekday()
+    days_since_last_monday = current_weekday if current_weekday != 0 else 7
+    last_monday = now - timedelta(days=days_since_last_monday)
+    last_monday_midnight = last_monday.replace(hour=0, minute=0, second=0, microsecond=0)
+    timestamp = int(last_monday_midnight.timestamp())
+    return timestamp
+
+
+@task
+def get_block_range() -> BlockRange:
```

Comment:
At the moment it executes the job once every 30 minutes, but it's very easy to change it. I think Haris recommended 3 hours so I can set it to that. Every time this job runs it re-uploads the entire data for that week, overwriting the current table (for that week). When we get to another week the start block will change and it will instead upload to a new table with a new name. 

So on a transition between weeks it will run the query multiple times for that day, and re-upload each time, changing the table when the startblock moves up a week.